Rocket.Chat/Managers/Model/SubscriptionManager/SubscriptionManager.swift

```diff
@@ -70,11 +70,20 @@ struct SubscriptionManager {
 
             currentRealm?.execute({ (realm) in
                 guard let auth = AuthManager.isAuthenticated(realm: realm), auth.serverURL == serverURL else { return }
-                let subscription = Subscription.getOrCreate(realm: realm, values: object, updates: { (object) in
-                    object?.auth = msg == "removed" ? nil : auth
-                })
 
-                realm.add(subscription, update: true)
+                let subscription: Subscription
+
+                if let rid = object["rid"].string, rid != "", let sub = Subscription.find(rid: rid) {
```

Comment:
`rid.isEmpty` instead of `!= ""`?

Reply:
This code could be a `guard` as well, instead of an `if`. I think makes more sense... what do you think?